core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileInStream.java

```diff
@@ -124,14 +130,17 @@ public int read(byte[] b, int off, int len) throws IOException {
           mPosition += bytesLeftInPage;
         } else { // cache miss
           byte[] page = readExternalPage(mPosition);
-          mCacheManager.put(pageId, page);
-          System.arraycopy(page, currentPageOffset, b, off + bytesRead, bytesLeftInPage);
-          bytesRead += bytesLeftInPage;
-          mPosition += bytesLeftInPage;
+          if (page.length > 0) {
+            mCacheManager.put(pageId, page);
+            System.arraycopy(page, currentPageOffset, b, off + bytesRead, bytesLeftInPage);
+            bytesRead += bytesLeftInPage;
+            mPosition += bytesLeftInPage;
+          }
         }
       }
     }
-    Preconditions.checkState(bytesRead == len, "Invalid number of bytes read");
+    Preconditions.checkState(bytesRead == len || (bytesRead < len && remaining() == 0),
+        "Invalid number of bytes read");
```

Comment:
please include numbers we are checking here in the error message, or it is hard to tell what's going wrong

Reply:
Done.